### PR TITLE
feat(trading): pane settings view

### LINKED ISSUE #5394: Pane settings view
## The Chore

Each pane might contain lots of different settings (for example column config, decimal precision). This ticket is to create a component or extend the components we have to allow each pane view to be toggled between the main view and the settings view. Ideally the toggle would exist in the pane context menu

Note that we don't need to populate the settings controls in this ticket, just provide the functionality to switch between views

## Tasks

- [x] Create pane settings ui and toggle
- [x] Dev
- [ ] Test

## Sketch

Screenshot 2023-11-29 at 18 30 46

### PR DESCRIPTION
**Pull request by bglownia**
# Related issues 🔗

Closes #5394 #5245

# Description ℹ️

Add option to reset tables filters and sorting in console app

# Demo 📺

image

### CODE REVIEW

File: libs/ui-toolkit/src/components/tabs/tabs.tsx [hunk 1]
```diff
@@ -122,6 +140,23 @@ export const Tabs = ({
           );
         })}
       </div>
+      {settingsOpened && !!settings ? (
```
Comment (mattrussell36) [inline libs/ui-toolkit/src/components/tabs/tabs.tsx:143]: For me this UI doesn't look tidy. I don't like the pane overlaying the tabs bar and how the button has to dropdown.

Comment (mattrussell36) [reply libs/ui-toolkit/src/components/tabs/tabs.tsx:143]: It needs a left border too.

File: libs/ui-toolkit/src/components/tabs/tabs.tsx [hunk 2]
```diff
@@ -38,12 +38,22 @@ export const Tabs = ({
     setWrapped(isWrapped(tabsRef.current, menuRef.current));
   });
 
+  const settings = children.find((c) => c?.props.id === activeTab)?.props
```
Comment (mattrussell36) [inline libs/ui-toolkit/src/components/tabs/tabs.tsx:41]: Why not just render the settings within the map of children, then you can access the settings render prop directly

File: libs/ui-toolkit/src/components/tabs/tabs.tsx [hunk 3]
```diff
@@ -18,12 +20,10 @@ export const Tabs = ({
   onValueChange,
   ...props
 }: TabsProps) => {
-  const [activeTab, setActiveTab] = useState<string | undefined>(() => {
-    if (defaultValue) {
-      return defaultValue;
-    }
-    return children.find((v) => v)?.props.id;
-  });
+  const [settingsOpened, setSettingsOpened] = useState(false);
```
Comment (mattrussell36) [inline libs/ui-toolkit/src/components/tabs/tabs.tsx:23]: I'm wondering if each pane store (the ones that use the `createDataGridSlice`) should have a `createSettingsSlice`, which includes a simple boolean field for if the settings are open and a setter function. This has some advantages such as being open the settings for a specific pane from anywhere. It also means we don't have to tie the tabs component to our 'pane' logic

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: libs/ui-toolkit/src/components/tabs/tabs.tsx (hunk 1 above)
Comment (mattrussell36) [reply libs/ui-toolkit/src/components/tabs/tabs.tsx:143]: Pr to merge here: https://github.com/vegaprotocol/frontend-monorepo/pull/5532

Comment (mattrussell36) [reply libs/ui-toolkit/src/components/tabs/tabs.tsx]: Resolving this as we are taking a different approach

File: libs/ui-toolkit/src/components/tabs/tabs.tsx (hunk 3 above)
Comment (bglownia) [reply libs/ui-toolkit/src/components/tabs/tabs.tsx]: Settings are in popover now.

**Review by daro-maj — APPROVED** [review]

**Review by mattrussell36 — APPROVED** [review]

**Pull request merged**